Python/optimizer.c

```diff
@@ -854,9 +855,13 @@ translate_bytecode_to_trace(
                                     operand = 0;
                                 }
                                 ADD_TO_TRACE(uop, oparg, operand, target);
+                                /* We need to guard that the runtime code object is
+                                 * the same one we are projecting into */
                                 code = new_code;
                                 func = new_func;
                                 instr = _PyCode_CODE(code);
+                                assert(code->co_version != 0);
+                                ADD_TO_TRACE(_GUARD_CODE, 0, code->co_version, 0);
```

Comment:
When could this be different, again? I forget.

If you have a reproducer, can you add it as a test?